source/communication_interfaces/include/communication_interfaces/sockets/TCPSocket.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <arpa/inet.h>
+
+#include "communication_interfaces/sockets/ISocket.hpp"
+
+namespace communication_interfaces::sockets {
+
+/**
+ * @brief Abstract class to define a generic TCP socket
+ * @details TCP is a connection-based communication protocol. Hence, TCP sockets need to implement an additional
+ * interface method `connect()`.
+ */
+class TCPSocket : public ISocket {
+public:
+  /**
+   * @brief Close the socket by calling TCPSocket::close()
+   */
+  ~TCPSocket() override;
+
+  /**
+   * @brief Connect the TCP socket to its counterpart (client / server)
+   * @details This method should be called after `open()` and may be blocking (for the case of a server)
+   */
+  virtual void connect() = 0;
```

Comment:
This does not respect the interface defined in ISocket. Would it be appropriate just to call `connect()` at the end of `open()`? Based on the simple test case it seems it would suffice.